Tests/PackageGraphTests/PubgrubTests.swift

```diff
@@ -801,6 +834,28 @@ final class PubgrubTests: XCTestCase {
         }
     }
 
+    func testResolutionWithBranchBasedDependency() {
```

Comment:
Can you add some more cases (including unsatisfiable ones) of branch-based dependencies?